nemo/collections/nlp/modules/common/megatron/token_level_encoder_decoder.py

```diff
@@ -226,55 +283,60 @@ def forward(
         """
         Return value is per token / per dimension (i.e., non collapsed loss value)
```

Comment:
A general question: does the PP limit returning a dict? Does this means that we need to run method multiple times to get more than a single value?

Reply:
Only torch tensors can be communicated across the pipeline. Only the last pipeline stage can return a dict (since it does not need to be communicated forward).